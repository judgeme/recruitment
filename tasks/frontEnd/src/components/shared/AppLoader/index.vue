<script setup lang="ts">
import { defineOptions } from 'vue';

// set display name in dev-tools
defineOptions({
  name: 'AppLoader'
});
</script>
<template>
	<span
		class="
			loader w-12 h-12 rounded-full 
			inline-block relative border-solid border-[3px]
			border-t-primary border-r-primary 
			border-b-transparent border-l-transparent
			after:absolute before:absolute 
			after:inset-0 before:inset-0
			after:m-auto before:m-auto 
			after:border-solid after:border-[3px]
			before:border-solid before:border-[3px] 
			after:border-t-transparent after:border-r-transparent
			after:border-b-primary after:border-l-primary
			after:w-10 after:h-10 after:rounded-full before:rounded-full
			before:w-8 before:h-8 before:border-t-primary
			before:border-r-primary
			before:border-b-transparent before:border-l-transparent
			animate-spin
		"
	>
	</span>
</template>
<style lang="css" scoped>
.loader::after,
.loader::before {
	content: '';
	animation: rotationBack 0.5s linear infinite;
	transform-origin: center center;
}
.loader::before {
	animation: rotation 1.5s linear infinite;
}

@keyframes rotation {
	0% {
		transform: rotate(0deg);
	}
	100% {
		transform: rotate(360deg);
	}
}
@keyframes rotationBack {
	0% {
		transform: rotate(0deg);
	}
	100% {
		transform: rotate(-360deg);
	}
}
</style>
