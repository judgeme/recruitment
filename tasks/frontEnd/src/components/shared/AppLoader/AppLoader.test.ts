import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import AppLoader from '@/components/shared/AppLoader/index.vue';

describe('AppLoader.vue', () => {
  it('matches the snapshot', () => {
    const wrapper = mount(AppLoader);
    expect(wrapper.html()).toMatchSnapshot();
  });
  
  it('should render correctly with the specified classes and styles', () => {
    const wrapper = mount(AppLoader);

    // Check if the component is rendered
    expect(wrapper.exists()).toBe(true);

    // Check the span element with the loader class
    const loaderSpan = wrapper.find('span.loader');
    expect(loaderSpan.exists()).toBe(true);

    // Check for the classes applied to the span element
    expect(loaderSpan.classes()).toContain('w-12');
    expect(loaderSpan.classes()).toContain('h-12');
    expect(loaderSpan.classes()).toContain('rounded-full');
    expect(loaderSpan.classes()).toContain('inline-block');
    expect(loaderSpan.classes()).toContain('relative');
    expect(loaderSpan.classes()).toContain('border-solid');
    expect(loaderSpan.classes()).toContain('border-[3px]');
    expect(loaderSpan.classes()).toContain('border-t-primary');
    expect(loaderSpan.classes()).toContain('border-r-primary');
    expect(loaderSpan.classes()).toContain('border-b-transparent');
    expect(loaderSpan.classes()).toContain('border-l-transparent');
    expect(loaderSpan.classes()).toContain('animate-spin');
  });
});
