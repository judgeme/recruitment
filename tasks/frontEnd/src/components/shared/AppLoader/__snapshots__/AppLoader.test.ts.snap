// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppLoader.vue > matches the snapshot 1`] = `"<span data-v-9e3594fe="" class="loader w-12 h-12 rounded-full inline-block relative border-solid border-[3px] border-t-primary border-r-primary border-b-transparent border-l-transparent after:absolute before:absolute after:inset-0 before:inset-0 after:m-auto before:m-auto after:border-solid after:border-[3px] before:border-solid before:border-[3px] after:border-t-transparent after:border-r-transparent after:border-b-primary after:border-l-primary after:w-10 after:h-10 after:rounded-full before:rounded-full before:w-8 before:h-8 before:border-t-primary before:border-r-primary before:border-b-transparent before:border-l-transparent animate-spin"></span>"`;
